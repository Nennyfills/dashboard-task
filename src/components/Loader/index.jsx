import PropTypes from 'prop-types';

const Loader = ({ cols, rows, notTable }) => {
  const shimmerRows = [];
  for (let count = 1; count <= rows; count += 1) {
    const shimmerCols = [];
    for (let countC = 1; countC <= cols; countC += 1) {
      shimmerCols.push(
        <td key={countC} data-testid="col">
          <div key={`shimmer-${countC}`} className="shimmer">
            <div className="shimmer-line" />
          </div>
        </td>
      );
    }
    shimmerRows.push(
      <tr key={count} data-testid="row">{shimmerCols && shimmerCols.map((col) => col)}</tr>
    );
  }
  if (notTable) {
    return (
      <div className="shimmer" data-testid="single-loader">
        <div className="shimmer-line" />
      </div>
    );
  }
  return shimmerRows && shimmerRows.map((shim) => shim);
};

Loader.propTypes = {
  rows: PropTypes.number,
  cols: PropTypes.number,
  notTable: PropTypes.bool,
};

Loader.defaultProps = {
  rows: 4,
  cols: 4,
  notTable: false,
};

export default Loader;
