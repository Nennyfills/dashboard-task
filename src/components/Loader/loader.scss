.center-loader {
  display: flex;
  flex-direction: column;
  align-content: center;
  justify-content: center;
  text-align: center;
  width: 100%;
}

.shimmer {
  animation-duration: 2.2s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: shimmer;
  animation-timing-function: linear;
  background: linear-gradient(
    to right,
    $light-grey-2 4%,
    #e0ecf3 25%,
    $light-grey 36%
  );
  background-size: 1200px 100%;
  border-radius: 4em;
  &-line {
    width: 100%;
    height: 15px;
    align-self: center;
    margin-left: 16px;
  }
}
@-webkit-keyframes shimmer {
  0% {
    background-position: -100% 0;
  }
  100% {
    background-position: 100% 0;
  }
}

@keyframes shimmer {
  0% {
    background-position: -1200px 0;
  }
  100% {
    background-position: 1200px 0;
  }
}
