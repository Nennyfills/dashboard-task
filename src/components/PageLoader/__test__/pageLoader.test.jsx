import { render, screen } from '@testing-library/react';
import Loader from '..';

it('Should take a snapshot', () => {
  const { asFragment } = render(<Loader />);
  expect(asFragment(<Loader />)).toMatchSnapshot();
});

it('Should render a PageLoader with test id page-loader', () => {
  render(<Loader />);
  const pageLoadClass = screen.getByTestId('page-loader');
  expect(pageLoadClass).toBeTruthy();
  expect(pageLoadClass).toHaveClass('page-loader');
});

it('Should render a PageLoader with test id page-loader-flex', () => {
  render(<Loader />);
  const pageLoadClass = screen.getByTestId('page-loader-flex');
  expect(screen.queryByTestId('page-loader-flex')).toBeTruthy();
  expect(pageLoadClass).toHaveClass('flex');
});

it('Should render a PageLoader with test id page-loader-override', () => {
  render(<Loader />);
  const pageLoadClass = screen.getByTestId('page-loader-override');
  expect(screen.queryByTestId('page-loader-override')).toBeTruthy();
  expect(pageLoadClass).toHaveClass('override');
});

it('Should render a PageLoader with test id page-loader-bottom', () => {
  render(<Loader />);
  const pageLoadClass = screen.getByTestId('page-loader-bottom');
  expect(screen.queryByTestId('page-loader-bottom')).toBeTruthy();
  expect(pageLoadClass).toHaveClass('loader');
});
