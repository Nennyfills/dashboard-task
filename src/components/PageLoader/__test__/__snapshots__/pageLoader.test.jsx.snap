// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="page-loader"
    data-testid="page-loader"
  >
    <div
      class="flex"
      data-testid="page-loader-flex"
    >
      <div>
        <div
          class="loader small-width"
        >
          <div
            class="shine"
          />
        </div>
        <div
          class="loader width"
        >
          <div
            class="shine"
          />
        </div>
      </div>
      <div
        class="flex override"
        data-testid="page-loader-override"
      >
        <div
          class="loader tiny-width"
        >
          <div
            class="shine"
          />
        </div>
        <div
          class="loader tiny-width"
        >
          <div
            class="shine"
          />
        </div>
        <div
          class="loader tiny-width"
        >
          <div
            class="shine"
          />
        </div>
        <div
          class="loader tiny-width"
        >
          <div
            class="shine"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="loader"
        data-testid="page-loader-bottom"
      >
        <div
          class="shine height"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
